src/calendar/use-calendar-grid-content.tsx

```diff
@@ -0,0 +1,48 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import React, { useMemo } from 'react';
+import styles from './styles.css.js';
+import clsx from 'clsx';
+import { normalizeStartOfWeek } from '../internal/utils/locale';
+import { getCalendarMonth } from 'mnth';
+import ScreenreaderOnly from '../internal/components/screenreader-only/index.js';
+import { renderDayName } from './utils/intl';
+
+export default function useCalendarGridContent({
```

Comment:
You're using `useMemo` here but it also doesn't really feel like a hook because you're mixing two things here.

Would it make sense to split this up into 1) a hook/utility to get rows, and 2) a pure functional component for the header ?

Reply:
I think this will make a bit more sense when granularity enters the scene. For month granularity there is no header, and it would be strange to have a function that returns `null` if `granularity` equals `month`. Also, note that `header` uses the first row in `rows` to generate its cell content. So there is some connection between the two, where different content for the grid is returned depending on the granularity.

In any case, we can always revisit this in #1942 soon.

Reply:
Why not splitting the hook into `useCalendarGridRows` hook and `CalendarGridHeader` component that takes rows as a prop?

Reply:
OK, split it as you suggested [here](https://github.com/cloudscape-design/components/pull/1963/commits/6a9a0dabf22fd7d5497720664143c0f00a84ffe1).